src/vellum/lib/test_suite.py

```diff
@@ -0,0 +1,175 @@
+                or metric_output.name == output_identifier
+            ]
+            for metric_output_execution in filtered_metric_outputs
+        ]
+
+        return filtered_metric_filtered_outputs
+
+
+ValidOutput = Union[
+    str,
+    float,
+    dict[str, Any],
+    list[SearchResult],
+    list[ChatMessage],
+    VellumError,
+]
+
+
+class VellumTestSuite:
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.client = Vellum(
+            api_key=os.environ.get("VELLUM_API_KEY"),
+        )
+        self.test_suite_id = self._get_test_suite_id_by_name(name)
+
+    def _get_test_suite_id_by_name(self, name: str) -> str:
+        # TODO: We need to EITHER:
+        # - Allow for all of these methods to be called with `name`
+        # - OR, we need to expose a get test suite id by name method and save the `id`
+        return uuid4()
+
+    def run_external(self, external_evaluator: Callable[..., dict[str, ValidOutput]]) -> VellumTestSuiteResults:
+        test_cases = self.client.test_suites.list_test_suite_test_cases(id=self.test_suite_id)
+        executions: list[ExternalTestCaseExecution] = []
+
+        # TODO - we need either:
+        # - Retrieve input_variables[i].key from test_suites
+        # - OR add a `name` field to `TestCaseVariableValue`
+        input_variable_name_by_id = {}
+
+        for test_case in test_cases.results:
+            inputs = test_case.input_values
+            external_input = {input_variable_name_by_id.get(inp.variable_id): inp.value for inp in inputs}
+            external_outputs = external_evaluator(
+                **external_input,
+            )
+
+            outputs = []
+            for external_output_key, external_output_value in external_outputs.items():
+                if isinstance(external_output_value, str):
+                    # Doesn't currently work - requires NamedTestCaseVariableValueRequest_String
+                    # Same for all other options below
+                    outputs.append(
+                        NamedTestCaseStringVariableValueRequest(
+                            value=external_output_value,
+                            name=external_output_key,
+                        )
+                    )
+
+                elif isinstance(external_output_value, float):
+                    outputs.append(
+                        NamedTestCaseNumberVariableValueRequest(
+                            value=external_output_value,
+                            name=external_output_key,
+                        )
+                    )
+
+                elif isinstance(external_output_value, dict):
+                    outputs.append(
+                        NamedTestCaseJsonVariableValueRequest(
+                            value=external_output_value,
+                            name=external_output_key,
+                        )
+                    )
+
+                # TODO: Handle the rest of the types
```

Comment:
highlighting this - the reason why we do the type/value paradigm is that it's hard to infer between list[SearchResult] and list[ChatMessage] for example. of JSON vs VellumError vs FunctionCall. My concern with the `dict` approach is that some if/elif inference chain will spread throughout our client sdk, making it harder to add new types. I propose we either:
- change the return type of `external_evaluator` to be a list of `NamedTestCaseVariableValue` instead of a dict
- change our API to accept either the list or dict, running the inference in a centralized location in our backend
- Stomach the concerns I listed above